File: leetcode/71.go
package leetcode

import (
	"bytes"
	"strings"
)
func simplifyPath(path string) string {
    s := strings.Split(path,"/")
    stk := make([]string,len(s))
    top := -1
    for i:=0;i<len(s);i++ {
        if len(s[i])==0 || s[i]=="." {
            continue
        }
        if s[i] == ".." {
            if top>-1 {
                top--
            }
            continue
        }
        top++
        stk[top] = s[i]
    }
    ans := bytes.Buffer{}
    ans.WriteString("/")
    for i:=0;i<top;i++ {
        ans.WriteString(stk[i])
        ans.WriteString("/")
    }
    if top!=-1 {
        ans.WriteString(stk[top])
    }
    return ans.String()
}

File: javascript/71.js
/**
 * @param {string} path
 * @return {string}
 */
 var simplifyPath = function(path) {
    // 维护一个栈就好了
    let stk = [];
    let s = path.split("/");
    s = s.filter(v=>{
        return v;
    });
    for(let x of s) {
        if(x==='.') {
            continue;
        } else if(x==='..') {
            if(stk.length)stk.pop();
        } else {
            stk.push(x);
        }
    }
    return "/" + stk.join("/");
};